systemtest/src/test/java/io/strimzi/systemtest/mirrormaker/MirrorMaker2IsolatedST.java

```diff
@@ -1129,6 +1121,28 @@ void testKMM2RollAfterSecretsCertsUpdateScramSha(ExtensionContext extensionConte
         LOGGER.info("Messages successfully mirrored");
     }
 
+    private static void modifyKafkaUserPasswordWithNewSecret(String namespace, String kafkaUserSourceName, String customSecretSource, String customPassword) {
+        Secret userDefinedSecret = new SecretBuilder()
+            .withNewMetadata()
+                .withName(customSecretSource)
+                .withNamespace(namespace)
+            .endMetadata()
+            .addToData("password", customPassword)
+            .build();
+        kubeClient().createSecret(userDefinedSecret);
+
+        KafkaUserResource.replaceUserResourceInSpecificNamespace(kafkaUserSourceName, ku -> {
+            ku.getSpec().getAuthentication()
+                .setAdditionalProperty(
+                    "password", Map.of(
+                        "valueFrom", Map.of(
+                            "secretKeyRef", Map.of(
+                                "name", customSecretSource,
+                                "key", "password"
+            ))));
```

Comment:
Wouldn't be `ku.getSpec().setAuthentication(new KafkaClientAuthenticationScramSha512Builder()...)` be much better than this?